### PR TITLE
HHH-15346 @ManyToOne associations not loaded correctly with default EAGER and batch fetch property set

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15346

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/BatchEntitySelectFetchInitializer.java [hunk 1]
```diff
@@ -147,7 +148,13 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
...
 
 		persistenceContext.getBatchFetchQueue().addBatchLoadableEntityKey( entityKey );
-		toBatchLoad.put( entityKey, parentAccess.getInitializedInstance() );
+		List<Object> objects = toBatchLoad.get( entityKey );
+		if ( objects == null ) {
+			objects = new ArrayList<>();
+		}
+		objects.add( parentAccess.getInitializedInstance() );
+
+		toBatchLoad.put( entityKey, objects );
```
Comment (frode-carlsen) [inline hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/BatchEntitySelectFetchInitializer.java:157]: this statement could be moved inside the if(objects==null) block

File: hibernate-core/src/test/java/org/hibernate/orm/test/batch/EagerManyToOneBatchTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+							.setParameter( "name", "a" )
+							.getResultList();
+
+					assertEquals( 2, children.size() );
+
+					Child child1 = children.get( 0 );
+					assertNotNull( child1.getParent() );
+
+					Child child2 = children.get( 1 );
+					assertNotNull( child2.getParent() );
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/batch/EagerManyToOneBatchTest.java:68]: Shouldn't we maybe also assert the number of SQL queries that are executed?

File: hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/BatchEntitySelectFetchInitializer.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/BatchEntitySelectFetchInitializer.java]: Thanks @frode-carlsen for the suggestion

**Review by beikov — APPROVED** [review]

File: hibernate-core/src/test/java/org/hibernate/orm/test/batch/EagerManyToOneBatchTest.java (hunk 2 above)
Comment (dreab8) [reply hibernate-core/src/test/java/org/hibernate/orm/test/batch/EagerManyToOneBatchTest.java]: I added the assertions and also an additional test class for a Lazy association

**Pull request merged**